code/modules/industrial_lift/tram_controls.dm

```diff
@@ -23,6 +26,17 @@
 	. = ..()
 	find_tram()
 
+	var/datum/lift_master/tram/tram_part = tram_ref?.resolve()
+	if(tram_part)
+		RegisterSignal(tram_part, COMSIG_TRAM_SET_TRAVELLING, .proc/set_controls_display)
+
+/obj/machinery/destination_sign/Destroy()
+	. = ..()
+
+	var/datum/lift_master/tram/tram_part = tram_ref?.resolve()
+	if(tram_part)
+		UnregisterSignal(tram_part, COMSIG_TRAM_SET_TRAVELLING)
```

Comment:
you do not need to do this, we autoclean signals.
Good thinking tho. just unneeded

Reply:
Removed.